external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaRDD.scala

```diff
@@ -60,6 +62,49 @@ class KafkaRDD[
     }.toArray
   }
 
+  override def count(): Long = offsetRanges.map(_.count).sum
```

Comment:
I thought about it, and I have a little concern about this. What if someone create a KafkaRDD with wrong offset ranges, which does not exist. In the current state, count will fail which is the correct thing to do. However, with this patch, it will give a count which is technically incorrect, rather than fail. May be a good idea to validate the limits of the offset ranges, to verify that they exist before returning the count. And do it just once.